client/src/connection_cache.rs

```diff
@@ -25,8 +25,8 @@ const DEFAULT_CONNECTION_CACHE_USE_QUIC: bool = true;
 /// construction of the ConnectionCache for code dealing both with udp and quic.
 /// For the scenario only using udp or quic, use connection-cache/ConnectionCache directly.
 pub enum ConnectionCache {
-    Quic(BackendConnectionCache<QuicPool, QuicConnectionManager, QuicConfig>),
-    Udp(BackendConnectionCache<UdpPool, UdpConnectionManager, UdpConfig>),
+    Quic(Arc<BackendConnectionCache<QuicPool, QuicConnectionManager, QuicConfig>>),
```

Comment:
Annotation: Arc is used so that we can Convert an Arc<ConnectionCache> to Arc<BackendConnectionCache>